internal/ioutil/io_test.go

```diff
@@ -77,52 +79,128 @@ func TestCopyBuffer(t *testing.T) {
-		args    args
-		wantDst string
-		wantErr error
+		name         string
+		args         args
+		wantDst      string
+		wantErr      error
+		blob         []byte
+		bufSize      int64
+		resumeOffset int64
 	}{
 		{
 			name:    "exact buffer size, no errors",
-			args:    args{bytes.NewReader(blob), make([]byte, 3), content.NewDescriptorFromBytes("test", blob)},
 			wantDst: "foo",
 			wantErr: nil,
+			blob:    blob,
+			bufSize: 3,
+		},
+		{
+			name:         "exact buffer size, no errors, resume",
+			wantDst:      "foo",
+			wantErr:      nil,
+			blob:         blob,
+			bufSize:      3,
+			resumeOffset: 1,
 		},
 		{
 			name:    "small buffer size, no errors",
-			args:    args{bytes.NewReader(blob), make([]byte, 1), content.NewDescriptorFromBytes("test", blob)},
 			wantDst: "foo",
 			wantErr: nil,
+			blob:    blob,
+			bufSize: 1,
+		},
+		{
+			name:         "small buffer size, no errors, resume",
+			wantDst:      "foo",
+			wantErr:      nil,
+			blob:         blob,
+			bufSize:      1,
+			resumeOffset: 1,
 		},
 		{
 			name:    "big buffer size, no errors",
-			args:    args{bytes.NewReader(blob), make([]byte, 5), content.NewDescriptorFromBytes("test", blob)},
 			wantDst: "foo",
 			wantErr: nil,
+			blob:    blob,
+			bufSize: 5,
+		},
+		{
+			name:         "big buffer size, no errors, resume",
+			wantDst:      "foo",
+			wantErr:      nil,
+			blob:         blob,
+			bufSize:      5,
+			resumeOffset: 1,
 		},
 		{
 			name:    "wrong digest",
-			args:    args{bytes.NewReader(blob), make([]byte, 3), content.NewDescriptorFromBytes("test", []byte("bar"))},
 			wantDst: "foo",
 			wantErr: content.ErrMismatchedDigest,
+			blob:    []byte("bar"),
+			bufSize: 3,
+		},
+		{
+			name:         "wrong digest, resume",
+			wantDst:      "foo",
+			wantErr:      content.ErrMismatchedDigest,
+			blob:         []byte("bar"),
+			bufSize:      3,
+			resumeOffset: 1,
 		},
 		{
 			name:    "wrong size, descriptor size is smaller",
-			args:    args{bytes.NewReader(blob), make([]byte, 3), content.NewDescriptorFromBytes("test", []byte("fo"))},
 			wantDst: "foo",
 			wantErr: content.ErrTrailingData,
+			blob:    []byte("fo"),
+			bufSize: 3,
+		},
+		{
+			name:         "wrong size, descriptor size is smaller, resume",
+			wantDst:      "foo",
+			wantErr:      content.ErrTrailingData,
+			blob:         []byte("fo"),
+			bufSize:      3,
+			resumeOffset: 1,
 		},
 		{
 			name:    "wrong size, descriptor size is larger",
-			args:    args{bytes.NewReader(blob), make([]byte, 3), content.NewDescriptorFromBytes("test", []byte("fooo"))},
 			wantDst: "foo",
 			wantErr: io.ErrUnexpectedEOF,
+			blob:    []byte("fooo"),
+			bufSize: 3,
 		},
+		// This case is, ugh, the whole point and it isn't verifying!!!
+		// {
+		// 	name:         "wrong size, descriptor size is larger, resume",
+		// 	wantDst:      "foo",
+		// 	wantErr:      io.EOF,
+		// 	blob:         []byte("fooo"),
+		// 	bufSize:      3,
+		// 	resumeOffset: 2,
+		// },
```

Comment:
Did you mean to leave this? Should this be passing?

Reply:
I forgot to finish it...getting the correct wantErr